### PR TITLE
fix(common): skip checking whether SVGs are oversized

### LINKED ISSUE #57941: Runtime warning NG0913 triggered for 300 kB svg file
### Which @angular/* package(s) are the source of the bug?

compiler, Don't known / other

### Is this a regression?

No

### Description

The runtime warning NG0913 is triggered for a 300 kB svg file. This file happens to have 'intrinsic' dimensions of > 7000 pixels, but in the case of svg files this is irrelevant.
This warning should not be triggered for svg images.

### Please provide a link to a minimal reproduction of the bug

_No response_

### Please provide the exception or error you saw

```true
NG0913: An image with src http://localhost:4200/assets/img/mandala-splash.svg has intrinsic file dimensions much larger than its rendered size. This can negatively impact application loading performance. For more information about addressing or disabling this warning, see https://angular.dev/errors/NG0913. Find more at https://angular.dev/errors/NG0913
```

### Please provide the environment you discovered this bug in (run `ng version`)

```true
Angular CLI: 18.2.5
Node: 20.14.0
Package Manager: npm 10.7.0
OS: darwin x64

Angular: 18.2.5
... animations, cli, common, compiler, compiler-cli, core, forms
... platform-browser, platform-browser-dynamic, router

Package                         Version
---------------------------------------------------------
@angular-devkit/architect       0.1802.5
@angular-devkit/build-angular   18.2.5
@angular-devkit/core            18.2.5
@angular-devkit/schematics      18.2.5
@angular/cdk                    18.2.4
@schematics/angular             18.2.5
rxjs                            7.8.1
typescript                      5.4.5
zone.js                         0.14.10
```

### Anything else?

_No response_

### PR DESCRIPTION
**Pull request by arturovt**
Prior to this commit, the `ImagePerformanceWarning` class was checking all `img` elements in the DOM to determine whether they were oversized after the DOM loading was complete. We should not check SVGs because they are vector-based and can scale infinitely without losing quality.

Closes #57941

### CODE REVIEW

File: packages/core/src/image_performance_warning.ts [hunk 1]
```diff
@@ -103,9 +103,30 @@ export class ImagePerformanceWarning implements OnDestroy {
...
+    const nonOversizedImageExtentions = [
+      // SVG images are vector-based, which means they can scale
+      // to any size without losing quality.
+      'svg',
+    ];
+
     images.forEach((image) => {
       if (!this.options?.disableImageSizeWarning) {
         for (const image of images) {
+          if (
```
Comment (JeanMeche) [inline packages/core/src/image_performance_warning.ts:120]: wdyt of moving everything into the `isOversized` method ?

File: packages/core/src/image_performance_warning.ts [hunk 2]
```diff
@@ -140,6 +140,28 @@ export class ImagePerformanceWarning implements OnDestroy {
...
+    ];
+
+    // Convert it to lowercase because this may have uppercase
+    // extensions, such as `IMAGE.SVG`.
+    // We fallback to an empty string because `src` may be `undefined`
+    // if it is explicitly set to `null` by some third-party code
+    // (e.g., `image.src = null`).
+    const imageSource = (image.src || '').toLowerCase();
+
+    if (nonOversizedImageExtentions.some((extension) => imageSource.endsWith(extension))) {
```
Comment (AndrewKushnir) [inline packages/core/src/image_performance_warning.ts:161]: Should we also add a `.` in front of an extension for better accuracy?
```suggestion
    if (nonOversizedImageExtentions.some((extension) => imageSource.endsWith('.' + extension))) {
```

**Review by AndrewKushnir — COMMENTED** [review]
@arturovt thanks for the PR, the code looks good 👍 Could you please also add a test to verify the new behavior and avoid regressions?

File: packages/core/src/image_performance_warning.ts (hunk 2 above)
Comment (arturovt) [reply packages/core/src/image_performance_warning.ts:161]: I replaced `svg` with `.svg`.

**Review by AndrewKushnir — APPROVED** [review]

**Comment by AndrewKushnir** [comment]
[Presubmit](http://test/OCL:679742523:BASE:679742580:1727474835713:293619f5).

**Review by JeanMeche — APPROVED** [review]

**Comment by AndrewKushnir** [comment]
@arturovt thanks for adding tests 👍 It looks like one of them is failing on CI (see `test` CI job):

```
packages/core/test/bundling/image-directive/index.ts:21:51 - error TS2307: Cannot find module './e2e/image-perf-warnings-oversized/svg-no-perf-oversized-warnings' or its corresponding type declarations.

21 import {SvgNoOversizedPerfWarningsComponent} from './e2e/image-perf-warnings-oversized/svg-no-perf-oversized-warnings';
                                                     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

Could you please take a look when you get a chance?

**Review by AndrewKushnir — APPROVED** [review]

**Comment by AndrewKushnir** [comment]
**Caretaker note**: presubmit is "green", this PR is ready for merge.

**Comment by alxhub** [comment]
This PR was merged into the repository by commit e8b2d5fad8aece3b2e3f4a7ed1356642f68ee275.

The changes were merged into the following branches: main, 18.2.x

**Pull request closed without merging**